interfaces/cython/cantera/ck2yaml.py

```diff
@@ -0,0 +1,2111 @@
+                        ('diameter', node.collision_diameter)])
+        if node.dipole_moment:
+            out['dipole'] = node.dipole_moment
+        if node.polarizability:
+            out['polarizability'] = node.polarizability
+        if node.z_rot:
+            out['rotational-relaxation'] = node.z_rot
+        if node.note:
+            out['note'] = node.note
+        return representer.represent_dict(out)
+
+
+def fortFloat(s):
+    """
+    Convert a string representation of a floating point value to a float,
+    allowing for some of the peculiarities of allowable Fortran representations.
+    """
+    s = s.strip()
+    s = s.replace('D', 'E').replace('d', 'e')
+    s = s.replace('E ', 'E+').replace('e ', 'e+')
+    return float(s)
+
+
+def get_index(seq, value):
+    """
+    Find the first location in *seq* which contains a case-insensitive,
+    whitespace-insensitive match for *value*. Returns *None* if no match is
+    found.
+    """
+    if isinstance(seq, str):
+        seq = seq.split()
+    value = value.lower().strip()
+    for i, item in enumerate(seq):
+        if item.lower() == value:
+            return i
+    return None
+
+
+def contains(seq, value):
+    if isinstance(seq, str):
+        return value.lower() in seq.lower()
+    else:
+        return get_index(seq, value) is not None
+
+
+class Surface:
+    def __init__(self, name, site_density):
+        self.name = name
+        self.site_density = site_density
+        self.species_list = []
+        self.reactions = []
+
+
+class Parser:
+    def __init__(self):
+        self.processed_units = False
+        self.energy_units = 'cal/mol'  # for the current REACTIONS section
+        self.output_energy_units = 'cal/mol'  # for the output file
+        self.quantity_units = 'mol'  # for the current REACTIONS section
+        self.output_quantity_units = 'mol'  # for the output file
+        self.motz_wise = None
+        self.warning_as_error = True
+
+        self.elements = []
+        self.element_weights = {}  # for custom elements only
+        self.species_list = []  # bulk species only
+        self.species_dict = {}  # bulk and surface species
+        self.surfaces = []
+        self.reactions = []
+        self.final_reaction_comment = ''
+        self.headerLines = []
+        self.files = []  # input file names
+
+    def warn(self, message):
+        if self.warning_as_error:
+            raise InputError(message)
+        else:
+            logging.warning(message)
+
+    def parse_composition(self, elements, nElements, width):
+        """
+        Parse the elemental composition from a 7 or 9 coefficient NASA polynomial
+        entry.
+        """
+        composition = {}
+        for i in range(nElements):
+            symbol = elements[width*i:width*i+2].strip()
+            count = elements[width*i+2:width*i+width].strip()
+            if not symbol:
+                continue
+            try:
+                count = int(float(count))
```

Comment:
Because [some people](https://github.com/Cantera/cantera/blob/master/test/data/nasa9-test-therm.dat) think you can write the atomic composition as a string like `4.00`, which `int` will choke on.

Reply:
Oy vey. This is why we have tests!